pkg/block/s3/adapter.go

```diff
@@ -120,6 +127,7 @@ func NewAdapter(ctx context.Context, params params.S3, opts ...AdapterOption) (*
 	a := &Adapter{
 		clients:             NewClientCache(cfg, params),
 		preSignedExpiry:     block.DefaultPreSignExpiryDuration,
+		preSignedEndpoint:   params.PreSignedEndpoint,
```

Comment:
Why is this initiated both here and via opts below? Seems redundant in that case.